src/table.ts

```diff
@@ -49,150 +50,65 @@ export interface File {
   generation?: number;
 }
 
-export interface JobMetadataCallback {
-  (err: Error|null, metadataOrResponse: JobMetadata|r.Response): void;
-}
-
+export type JobMetadataCallback = RequestCallback<JobMetadata>;
 export type JobMetadataResponse = [JobMetadata];
 
 // tslint:disable-next-line no-any
 export type RowMetadata = any;
 
-export interface InsertRowsOptions {
+export type InsertRowsOptions = bigquery.IInsertAllTableDataRequest&{
   autoCreate?: boolean;
-  ignoreUnknownValues?: boolean;
   raw?: boolean;
-  schema?: string|{};
-  skipInvalidRows?: boolean;
-  templateSuffix?: string;
-}
-
-export type RowsResponse = [RowMetadata[], GetRowsOptions, r.Response];
-export interface RowsCallback {
-  (err: Error|null, rows?: RowMetadata[]|null, nextQuery?: GetRowsOptions|null,
-   apiResponse?: r.Response): void;
-}
-
-export interface TableRow {
-  f: TableRowField[];
-}
-export interface TableRowField {
-  v: string|TableRow|TableRowField[];
-}
+  schema: string|{};
+};
+
+// @TODO when breaking changes are cool, rename these to have suffix InsertRows
+export type ApiResponse =
+    [bigquery.IInsertAllTableDataResponse | bigquery.ITable];
+export type ApiResponseCallback =
+    RequestCallback<bigquery.IInsertAllTableDataResponse|bigquery.ITable>;
+
+export type RowsResponse = PagedResponse<
+    RowMetadata, GetRowsOptions,
+    bigquery.IListTableDataResponse|bigquery.ITable>;
+export type RowsCallback = PagedCallback<
+    RowMetadata, GetRowsOptions,
+    bigquery.IListTableDataResponse|bigquery.ITable>;
+
+export type TableRow = bigquery.ITableRow;
+export type TableRowField = bigquery.ITableCell;
 export type TableRowValue = string|TableRow;
 
-export interface GetRowsOptions {
-  startIndex?: number;
-  selectedFields?: string;
-  autoPaginate?: boolean;
-  maxApiCalls?: number;
-  maxResults?: number;
-}
+export type GetRowsOptions = PagedRequest<bigquery.IListTableDataRequest>;
 
-export interface JobLoadMetadata {
-  jobId?: string;
-  jobPrefix?: string;
-  allowJaggedRows?: boolean;
-  allowQuotedNewlines?: boolean;
-  autodetect?: boolean;
-  clustering?: {fields: number[];};
-  createDisposition?: string;
-  destinationEncryptionConfiguration?: {kmsKeyName?: string;};
-  destinationTable?: {datasetId: string; projectId: string; tableId: string;};
-  destinationTableProperties?: {description?: string; friendlyName?: string;};
-  encoding?: string;
-  fieldDelimiter?: string;
-  ignoreUnknownValues?: boolean;
-  maxBadRecords?: number;
-  nullMarker?: string;
-  projectionFields?: string[];
-  quote?: string;
-  schema?: {fields: TableField[]};
-  schemaUpdateOptions?: string[];
-  skipLeadingRows?: number;
-  sourceFormat?: string;
+export type JobLoadMetadata = JobRequest<bigquery.IJobConfigurationLoad>&{
   format?: string;
```

Comment:
I'm relatively new to TypeScript (please excuse any silly questions I have); why do we continue defining `format?`; does this mean that `bigquery.IJobConfigurationLoad` already encompasses all the old types, but we're extending on it to introduce `format`?

Reply:
Yep, thats correct. Sometimes we offer convenience options and I believe this is one of them